packages/headless/src/features/facets/category-facet-set/category-facet-set-slice.ts

```diff
@@ -44,22 +44,29 @@ export const categoryFacetSetReducer = createReducer(
           return;
         }
 
-        state[facetId] = buildCategoryFacetRequest(options);
+        state[facetId] = {
+          request: buildCategoryFacetRequest(options),
+        };
       })
       .addCase(change.fulfilled, (_, action) => action.payload.categoryFacetSet)
       .addCase(restoreSearchParameters, (state, action) => {
         const cf = action.payload.cf || {};
         const facetIds = Object.keys(state);
 
         facetIds.forEach((id) => {
-          const request = state[id];
+          const request = state[id]?.request;
+
+          if (!request) {
+            return;
+          }
```

Comment:
Since `id` was obtained from `Object.keys(state)`, is this condition necessary? A `!.` operator could be used above instead.

Reply:
Yes, good point!